crates/bevy_math/src/rects/urect.rs

```diff
@@ -284,17 +287,21 @@ impl URect {
     ///
     /// ```rust
     /// # use bevy_math::{URect, UVec2};
-    /// let r = URect::new(2, 2, 4, 4); // w=2 h=2
+    /// let r = URect::new(4, 4, 6, 6); // w=2 h=2
     /// let r2 = r.inset(1); // w=3 h=3
-    /// assert_eq!(r2.min, UVec2::splat(1));
-    /// assert_eq!(r2.max, UVec2::splat(5));
+    /// assert_eq!(r2.min, UVec2::splat(3));
+    /// assert_eq!(r2.max, UVec2::splat(7));
+    ///
+    /// let r = URect::new(4, 4, 8, 8); // w=4 h=4
+    /// let r2 = r.inset(-1); // w=3 h=3
+    /// assert_eq!(r2.min, UVec2::splat(5));
+    /// assert_eq!(r2.max, UVec2::splat(7));
     /// ```
     #[inline]
-    pub fn inset(&self, inset: u32) -> Self {
-        assert!(self.min.cmpge(UVec2::splat(inset)).all(), "Inset must be smaller than this rect's min otherwise the rectangle is undefined! Inset was {inset} and this rect's min was {}", self.min);
+    pub fn inset(&self, inset: i32) -> Self {
         let mut r = Self {
-            min: self.min - inset,
-            max: self.max + inset,
+            min: (self.min.as_ivec2() - inset).as_uvec2(),
```

Comment:
I'd rather avoid the conversion to signed, although it's a bit unlikely it still loses some precision and will return wrong results for many `u32` values which can't fit into an `i32`.

Reply:
Do you think `expand` and `shrink` methods for `URect` is the correct solution?

Reply:
Maybe, although that makes the API different from `IRect` and `Rect` which is probably confusing. But you can just calculate the new size here without conversion, just need to fall back to per-component operation because there's no good way to do it directly with the `UVec2` interface I think.

Also about naming I'd expect `expand` to expand the current object in-place, vs. `expanded` to return a new expanded object.